### PR TITLE
feat(@angular/cli): add barebones option to ng new

### PR DESCRIPTION
**Pull request by Brocco**
This feature allows users to specify `--barebones` (`-b`) to create a no-frills app.

### CODE REVIEW

**Comment by StephenFluin** [comment]
It would be nice to remove the environments files and the favicon, but I think it's fine with those left in.

File: packages/@angular/cli/blueprints/ng/index.ts [hunk 1]
```diff
@@ -78,6 +91,17 @@ export default Blueprint.extend({
...
+    if (this.options && this.options.barebones) {
+      fileList = fileList.filter(p => p.indexOf('e2e/') < 0);
+      fileList = fileList.filter(p => p.indexOf('editorconfig') < 0);
+      fileList = fileList.filter(p => p.indexOf('README.md') < 0);
+      fileList = fileList.filter(p => p.indexOf('karma.conf.js') < 0);
+      fileList = fileList.filter(p => p.indexOf('protractor.conf.js') < 0);
+      fileList = fileList.filter(p => p.indexOf('test.ts') < 0);
+      fileList = fileList.filter(p => p.indexOf('tsconfig.spec.json') < 0);
+      fileList = fileList.filter(p => p.indexOf('tslint.json') < 0);
+    }
```
Comment (gioragutt) [inline packages/@angular/cli/blueprints/ng/index.ts]: Another way is to save all those files in a list, and use `fileList = fileList.filter(p => blackList.indexOf(p) < 0);`

File: packages/@angular/cli/blueprints/ng/index.ts [hunk 2]
```diff
@@ -78,6 +91,18 @@ export default Blueprint.extend({
       fileList = fileList.filter(p => p.indexOf('app.component.spec.ts') < 0);
     }
 
+    if (this.options && this.options.barebones) {
+      fileList = fileList.filter(p => p.indexOf('e2e/') < 0);
```
Comment (hansl) [inline packages/@angular/cli/blueprints/ng/index.ts]: Can you chain those and use Regexes? I think it'd be simpler to read.

```typescript
fileList = fileList.filter(p => {
  return !(   p.match(/e2e\//)
           || p.match(/editorconfig/)
           || p.match(/README\.md/)
           || /* ... */)
```

Opened to suggestions, I just don't think `indexOf` here is not that legible.

Comment (hansl) [inline packages/@angular/cli/blueprints/ng/index.ts]: Or even better, have a list of files to remove and filter those out:

```typescript
const toRemoveList = [/e2e\//, /editorconfig/, /README/, /* ... */];
fileList = fileList.filter(p => {
  return !toRemoveList.some(re => p.match(re));
});```

**Pull request merged**